File: 2. Data Structers And Algorithms/3. Stack/Stack Data Structer/LinkedList.cs
using System.Numerics;

namespace Stack_Data_Structer
{
    class Node
    {
        internal int value;
        internal Node next = null;
        public Node() {}
        public Node(int value) { this.value = value; }
    }
    internal class LinkedList
    {
        private Node head = null;
        private int length;

        public LinkedList()
        {
        }

        public bool IsEmpty() => this.length == 0 || this.head == null;
        public int Length() => this.length;
        public void PushHead(int value)
        {
            Node node = new Node(value);
            if (length == 0)
            {
                this.head = node;

            }
            else
            {
                node.next = this.head;
                this.head = node;
            }
            this.length++;
        }
        public void PopHead()
        {
            if (length == 0)
            {
                return;
            }
            else
            {
                this.head = this.head.next;
            }
            this.length--;
        }
        public int Head()
        {
            if(this.head == null)
            {
                return int.MinValue;
            }else
            {
                return this.head.value;
            }
        }

    }
}

File: 2. Data Structers And Algorithms/4. Queue/Queue Implement By LinkedList/src/struct/LinkedList.java
package struct;

public class LinkedList {
	private Node head;
	private Node tail;
	
	public LinkedList(){
		head = null;
		tail = null;
	}
	
	public boolean isEmpty(){
		return head == null || tail == null;
	}
	
	public Node getHead(){
		return head;
	}
	
	public Node getTail(){
		return tail;
	}
	
	public void pushTail(int _data){
		Node node = new Node(_data);
		if(isEmpty()){
			head = tail = node;
		} else {
			tail.next = node;
			tail = node;
		}
	}
	
	public Node popHead(){
		if(isEmpty()) return null;
		Node node = this.head;
		this.head = this.head.next;
		return node;
	}
	
	
}
